pkg/engine/anchor/common.go

```diff
@@ -125,3 +138,46 @@ func RemoveAnchorsFromPath(str string) string {
 func AddAnchor(key, anchorPrefix string) string {
 	return anchorPrefix + key + ")"
 }
+
+// IsConditionAnchor checks for condition anchor
+func IsConditionAnchor(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsConditionAnchor()
+}
+
+// IsGlobalAnchor checks for global condition anchor
+func IsGlobalAnchor(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsGlobalAnchor()
+}
+
+// ContainsCondition returns true, if str is either condition anchor or
+// global condition anchor
+func ContainsCondition(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsConditionAnchor() || ah.IsGlobalAnchor()
+}
+
+// IsNegationAnchor checks for negation anchor
+func IsNegationAnchor(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsNegationAnchor()
+}
+
+// IsAddIfNotPresentAnchor checks for addition anchor
+func IsAddIfNotPresentAnchor(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsAddIfNotPresentAnchor()
+}
+
+// IsEqualityAnchor checks for equality anchor
+func IsEqualityAnchor(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsEqualityAnchor()
+}
+
+// IsExistenceAnchor checks for existence anchor
+func IsExistenceAnchor(str string) bool {
+	ah := ParseAnchor(str)
+	return ah.IsExistenceAnchor()
+}
```

Comment:
can't we drop all those funcs ?

Reply:
right now no, because of this https://github.com/kyverno/kyverno/pull/6133#issuecomment-1406649444